### ISSUE TITLE
Flaky test: OpenCensusIntegration

### ISSUE THREAD

**Issue by easwars** [grpc/grpc-go#5559]
```
panic: Log in goroutine after Test/OpenCensusIntegration has completed: Metrics[grpc.io/client/received_bytes_per_rpc]

goroutine 17 [running]:
testing.(*common).logDepth(0x4000003860, {0x4000346090, 0x2e}, 0x3)
	/opt/hostedtoolcache/go/1.18.4/x64/src/testing/testing.go:887 +0x3d4
testing.(*common).log(...)
	/opt/hostedtoolcache/go/1.18.4/x64/src/testing/testing.go:869
testing.(*common).Logf(0x4000003860, {0x8a0cb5?, 0x40001001b0?}, {0x40004f6d60?, 0xffffffffffffffff?, 0x0?})
	/opt/hostedtoolcache/go/1.18.4/x64/src/testing/testing.go:920 +0x5c
google.golang.org/grpc/gcp/observability.(*fakeOpenCensusExporter).ExportView(0x40001baba0, 0x40002fe000)
	/home/runner/work/grpc-go/grpc-go/gcp/observability/observability_test.go:371 +0x100
go.opencensus.io/stats/view.(*worker).reportView(0x40001d8080, 0x4000248[320](https://github.com/grpc/grpc-go/runs/7635718800?check_suite_focus=true#step:8:321))
	/home/runner/go/pkg/mod/go.opencensus.io@v0.23.0/stats/view/worker.go:357 +0x204
go.opencensus.io/stats/view.(*worker).reportUsage(0x40001d8080)
	/home/runner/go/pkg/mod/go.opencensus.io@v0.23.0/stats/view/worker.go:365 +0xac
go.opencensus.io/stats/view.(*worker).start(0x40001d8080)
	/home/runner/go/pkg/mod/go.opencensus.io@v0.23.0/stats/view/worker.go:280 +0xbc
created by go.opencensus.io/stats/view.init.0
	/home/runner/go/pkg/mod/go.opencensus.io@v0.23.0/stats/view/worker.go:34 +0xa8
FAIL	google.golang.org/grpc/gcp/observability	0.948s
?   	google.golang.org/grpc/gcp/observability/internal/config	[no test files]
?   	google.golang.org/grpc/gcp/observability/internal/logging	[no test files]
FAIL
```

https://github.com/grpc/grpc-go/runs/7635718800?check_suite_focus=true

**Comment by easwars** [comment]
@zasweq

**Comment by zasweq** [comment]
Fixed by #5623.
